pkg/controller/controller_binding.go

```diff
@@ -506,6 +531,7 @@ func (c *controller) reconcileServiceBinding(binding *v1beta1.ServiceBinding) er
 					v1beta1.ConditionFalse,
 					errorBindCallReason,
 					"Bind call failed. "+s)
+				binding.Status.UnbindStatus = v1beta1.ServiceBindingUnbindStatusNotRequired
```

Comment:
The Controller should not be transitioning the `UnbindStatus` back to `NotRequired`. Even if a bind request fails, the Controller still needs to send an unbind request for the ServiceBinding. See #1541.

I haven't yet been able to convince myself about what we should do in the case of having a non-bindable plan after a bind request was already sent. Is that even a reasonable scenario?